prog/vnet/subnet_nexus.rb

```diff
@@ -38,13 +38,90 @@ def wait
       hop :refresh_mesh
     end
 
+    when_refresh_keys_set? do
+      ps.update(state: "refreshing_keys")
+      hop :refresh_keys
+    end
+
     nap 30
   end
 
+  def gen_encryption_key
+    "0x" + SecureRandom.bytes(36).unpack1("H*")
+  end
+
+  def gen_spi
+    "0x" + SecureRandom.bytes(4).unpack1("H*")
+  end
+
+  def gen_reqid
+    SecureRandom.random_number(100000) + 1
```

Comment:
It'd probably more idiomatic to do 

```suggestion
    SecureRandom.random_number(1..100000)
```